projects/storefrontlib/cms-components/misc/icon/icon-loader.service.ts

```diff
@@ -23,69 +24,76 @@ export class IconLoaderService {
 
   /**
    * Returns an html fragment which can be added to the DOM in a safe way.
+   *
+   * @deprecated only exists for backwards compatibility, html generation has been moved to icon component
    */
-  getHtml(type: ICON_TYPE | string): SafeHtml | undefined {
-    if (this.isResourceType(type, IconResourceType.SVG)) {
-      return this.sanitizer.bypassSecurityTrustHtml(
-        `<svg><use xlink:href="${this.getSvgPath(type)}"></use></svg>`
+  public getHtml(type: ICON_TYPE_STRING): SafeHtml | undefined {
+    if (this.getResourceType(type) === IconResourceType.SVG) {
+      const url = this.sanitizer.sanitize(
+        SecurityContext.URL,
+        this.getSvgPath(type) || null
       );
+      if (url) {
+        const useElement = this.winRef.document.createElement('use');
+        useElement.setAttribute('xlink:href', url);
+        const svgElement = this.winRef.document.createElement('svg');
+        svgElement.appendChild(useElement);
+        return this.sanitizer.bypassSecurityTrustHtml(svgElement.outerHTML);
+      }
     }
-    if (this.isResourceType(type, IconResourceType.TEXT)) {
+    if (this.getResourceType(type) === IconResourceType.TEXT) {
       const symbol = this.getSymbol(type);
       if (symbol) {
-        return this.sanitizer.bypassSecurityTrustHtml(symbol);
+        const helperDiv = this.winRef.document.createElement('div');
+        helperDiv.textContent = symbol;
+        return this.sanitizer.bypassSecurityTrustHtml(helperDiv.innerHTML);
       }
     }
   }
 
+  /**
+   * Get the `IconResourceType` that is configured for the given `ICON_TYPE`
+   * Defaults to `IconResourceType.LINK`, if an unknown `IconResourceType` is configured or none is configured.
+   */
+  public getResourceType(iconType: ICON_TYPE_STRING): IconResourceType {
+    const iconResourceType = this.config?.resources?.find((res) =>
+      res.types?.includes(iconType)
+    )?.type;
+    if (!Object.values<any>(IconResourceType).includes(iconResourceType)) {
+      return IconResourceType.LINK;
+    }
```

Comment:
Looks like the old implementation defaulted to `LINK`. Or should we fail differently? In particular, when the config contains an unexpected IconResourceType?

Btw, I hope there is a better way to phrase this `if`-condition in TypeScript. But I couldn't figure it out. I tried to change the return type to `keyof typeof IconResourceType`, and let TypeScript do the rest. But that didn't work, because we need the values of the enum, not the keys.

Reply:
Let's not return a different resource type unless there is a compelling reason to do so.  That would be a non backwards compatible change.